### PR TITLE
New method Find Column w.r.t. #1619

### PR DESCRIPTION
**Pull request by atishamte**
W.r.t #1619, correction in the codes and introduced test cases as well

Its a shorthand query.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by lonnieezell** [comment]
I believe the purpose of the originally proposed method was to return an array of the values only, at least when a single column was asked for, like: 

```
$job = $model->findColumn('name');
$this->assertEquals('Developer', $job[0]);
$this->assertEquals('Politician', $job[1]);
$this->assertEquals('Accountant', $job[2]);
$this->assertEquals('Musician', $job[3]);
```

That makes it easy to grab a column of values and then immediately use it for a whereIn, or whatever.

**Comment by atishamte** [comment]
I kept the purpose is to shorthand the query, where you can get the response in the format of object or array when you can pass single and multiple columns.
But the response format will remain the same.
If I make any changes w.r.t. only array and someone wants a response in an object, then parsing is the only option to do so.
Also, there would be a confusion of the return type of single and multidimensional array w.r.t. single and multiple columns respectively.

**Comment by lonnieezell** [comment]
My personal feeling is that if all it does is provide a shorthand way to get rid of doing `->select('x')` then it's not a feature I'm interested in. Returning an array of values is something I can see being a time saver for a lot of people so I'm ok with that. 

With that in mind - I think it should always return an array of the actual values. If need be, restrict it to a single column name.

**Comment by atishamte** [comment]
No problem. I will make the necessary changes

**Comment by atishamte** [comment]
Does the second parameter is required in the method? or we should strictly keep it as a result set of all values under the column without where condition.

**Comment by atishamte** [comment]
I have done the changes for single column input along with exception handling

**Comment by atishamte** [comment]
@lonnieezell and @jim-parry, any changes required in this PR?

File: system/Model.php [hunk 1]
```diff
@@ -349,6 +349,42 @@ public function find($id = null)
...
+		                  ->find();
+
+		if (count($resultSet))
+		{
+			$data = [];
+			foreach ($resultSet as $item)
+			{
+				$data[] = $item[$columnName];
+			}
+
```
Comment (lonnieezell) [inline system/Model.php]: Using `array_column` is likely faster here.

**Comment by atishamte** [comment]
Changes done according to array_column

**Comment by atishamte** [comment]
@lonnieezell any changes required here?

**Comment by lonnieezell** [comment]
Looks good! Thanks.

**Pull request merged**